### PR TITLE
Support errorprone being enabled by setting `com.palantir.baseline-error-prone.disable=false`

### PR DESCRIPTION
**Pull request by CRogers**
## Before this PR
This behaviour has changed from baseline, which was [this snippet](https://github.com/palantir/gradle-baseline/blob/addba397fceeced9d284c715b9588257afa6f38e/gradle-baseline-java/src/main/java/com/palantir/baseline/plugins/BaselineErrorProne.java#L327-L334).

Causing a test to fail on https://github.com/palantir/gradle-baseline/pull/2944

## After this PR

==COMMIT_MSG==
Ensure we match the behaviour of `com.palantir.baseline-error-prone.disable` from baseline.
==COMMIT_MSG==

### CODE REVIEW

File: gradle-suppressible-error-prone/src/main/java/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePlugin.java [hunk 1]
```diff
@@ -285,4 +287,13 @@ static String excludedPathsRegex() {
...
     }
+
+    private static boolean isDisabledViaLegacyBaselineProperty(Project project) {
+        Object disable = project.findProperty(ERROR_PRONE_BASELINE_DISABLE);
+        if (disable == null) {
+            return false;
+        } else {
+            return !disable.equals("false");
+        }
+    }
```
Comment (felixdesouza) [inline gradle-suppressible-error-prone/src/main/java/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePlugin.java]: nit:

```
disable != null && !disable.equals("false")
```

?

File: gradle-suppressible-error-prone/src/test/groovy/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePluginIntegrationTest.groovy [hunk 2]
```diff
@@ -449,7 +449,7 @@ class SuppressibleErrorPronePluginIntegrationTest extends IntegrationSpec {
     }
 
     def 'can disable errorprone using property'() {
-        when:
+        when: 'there is some that will fail an errorprone'
```
Comment (felixdesouza) [inline gradle-suppressible-error-prone/src/test/groovy/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePluginIntegrationTest.groovy]: not quite sure what you mean here

**Review by felixdesouza — DISMISSED** [review]
some nits, removed `merge when ready`

File: gradle-suppressible-error-prone/src/test/groovy/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePluginIntegrationTest.groovy (hunk 2 above)
Comment (CRogers) [reply gradle-suppressible-error-prone/src/test/groovy/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePluginIntegrationTest.groovy]: improved the comment

**Review by felixdesouza — APPROVED** [review]

**Pull request merged**